### ISSUE TITLE
chore(v3): set UTF-8 default-on

### ISSUE THREAD

**Issue by ywwg** [prometheus/prometheus#14674]
### What did you do?

per https://github.com/prometheus/prometheus/pull/14490, UTF-8 is default-off in the main branch.  When that change is pulled into the v3 branch, we need to apply  6d2747dbec0fdc61a4449b673acdf47faa09f487 and some fixes in order to flip the default to on.

### What did you expect to see?

n/a

### What did you see instead? Under which circumstances?

n/a

### Prometheus version

```text
HEAD
```

### Prometheus configuration file

```yaml
n/a
```

### Alertmanager version

```text
n/a
```

### Alertmanager configuration file

```yaml
n/a
```

### Logs

```text
n/a
```

**Comment by ArthurSens** [comment]
Since the next release is v3, couldn't we merge this into main right now?

**Comment by ywwg** [comment]
Per the discussion in the PR, I was asked not to

**Comment by ArthurSens** [comment]
> Per the discussion in the PR, I was asked not to

Oh apologies, I should have read the discussion over there first 😬

**Comment by jan--f** [comment]
Adding the `breaking` label. Iiuc this could change functionality so that user setups break if they upgrade without further action. Feel free to remove if this isn't that case.

**Comment by ywwg** [comment]
Yes I agree this is a breaking change.  It is possible to keep the old behavior but that requires config changes

**Comment by ywwg** [comment]
done!
